### PR TITLE
angular-cli 6 (with webpack 4) compatibility

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: #3399 

## What I did
- [x] Upgraded to v6 of the Angular related deps 
- [x] Adopt migration of the angular-cli.json -> angular.json
- [x] Update peers to >=6

CC. @Quramy  @avol-io @bengry

### CODE REVIEW

File: app/angular/src/server/angular-cli_config.js [hunk 1]
```diff
@@ -21,12 +21,19 @@ export function getAngularCliWebpackConfigOptions(dirToSearch, appIndex = 0) {
     throw new Error('.angular-cli.json must have apps entry.');
   }
   const appConfig = cliConfig.apps[appIndex];
+  const { root, styles, scripts } = appConfig;
```
Comment (Quramy) [inline app/angular/src/server/angular-cli_config.js]: We should read the build options from `angular.json` (not `.angular-cli.json`). And the structures of them are not compatible.

```js
const angularJson = JSON.parse(fs.readFileSync(path.resolve(dirToSearch, 'angular.json'), 'utf8'))
const project = angularJson['projects'][angularJson['defaultProject']];
const cliWebpackConfigOptions = {
  buildOptions: project.architect.build.options,
  root: project.root,
  projectRoot: dirToSearch,
  supportES2015: false,
  tsConfig: {
    options: { },
    fileNames: [],
    errors: [],
  },
  tsConfigPath: path.resolve(dirToSearch, 'src/tsconfig.app.json'),
};
```

I've created the sample repo to extract webpack config object using `@angular-devkit/build-angular` 
https://github.com/Quramy/how-to-extract-webpack-conf-from-angular-cli

File: examples/angular-cli/package.json [hunk 2]
```diff
@@ -13,20 +13,22 @@
...
   },
   "dependencies": {
-    "@angular/common": "^5.2.10",
-    "@angular/compiler": "^5.2.10",
-    "@angular/core": "^5.2.10",
-    "@angular/forms": "^5.2.10",
-    "@angular/platform-browser": "^5.2.10",
-    "@angular/platform-browser-dynamic": "^5.2.10",
-    "@ngrx/store": "^5.2.0",
+    "@angular/common": "^6.0.0-rc.5",
```
Comment (Quramy) [inline examples/angular-cli/package.json]: As mentioned above, new angular/cli v6 needs not `.angular-cli.json` but `angular.json`. I think it's better to re-generate the example project with `ng new angular-cli`.

**Review by Quramy — CHANGES_REQUESTED** [review]

File: examples/angular-cli/package.json (hunk 2 above)
Comment (Hypnosphi) [reply examples/angular-cli/package.json]: Stable 6 version is out now, please upgrade

**Comment by igor-dv** [comment]
@Quramy, I have a strange thing with "assets" prop.

I was needed to override it like this:

```js
buildOptions: {
      ...project.architect.build.options,
      assets: [],
    },
```

Otherwise, I am getting an error from this code:

image

which is strange because assets were generated by angular-cli and they are:
```json
 "assets": [
              "src/favicon.ico",
              "src/assets"
            ]
```

**Comment by igor-dv** [comment]
@klimentru1986, after your angular PRs, I assume you can help here =)

**Comment by daiky00** [comment]
is this going to get fixed?

**Comment by ndelangen** [comment]
What's the status of this @igor-dv ?

**Comment by igor-dv** [comment]
Something bad happens with the static build after the merge from master 🤔

**Review by ndelangen — APPROVED** [review]

File: app/angular/src/server/angular-cli_utils.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  }
+}
+
+export function normalizeAssetPatterns(assetPatterns, dirToSearch, sourceRoot) {
+  if (!assetPatterns || !assetPatterns.length) {
+    return [];
+  }
+
+  return assetPatterns.map(assetPattern => {
+    const assetPath = normalize(assetPattern);
```
Comment (santialbo-actimo) [inline app/angular/src/server/angular-cli_utils.js]: Hi @igor-dv sorry to nudge you like this. I'm gettinga an error here.
At this point my asset pattern looks like this:
```
{ glob: '**/*',
  input: 'apps/viewer/src/assets',
  output: '/assets' }
```

which throws the following error:
```
TypeError: path.match is not a function
```

Comment (igor-dv) [reply app/angular/src/server/angular-cli_utils.js]: Thanks, will take a look

**Comment by FacePalmDev** [comment]
I'm not sure what's happening with this issue? Is it being looked at? Is it resolved?

**Pull request merged**